src/map/battleground.cpp

```diff
@@ -222,59 +621,863 @@ int bg_team_get_id(struct block_list *bl)
+
+					if (pl_sd && list.size() < bg->required_players)
+						list.push_back(pl_sd);
+				}
+			}
+
+			return bg_queue_join_multi(name, sd, list); // Join as guild, all on the same side of the BG
+		} else {
+			ShowWarning("clif_parse_bg_queue_apply_request: Could not find Battleground: \"%s\" requested by player: %s (AID:%d CID:%d)\n", name, sd->status.name, sd->status.account_id, sd->status.char_id);
+			return BG_APPLY_INVALID_NAME; // Invalid BG name
+		}
+	}
+}
+
+/**
+ * Join multiple players onto the same side of a Battleground
+ * @param name: Battleground name
+ * @param sd: Player who requested to join the battlegrounds
+ * @param list: Contains all players including the player who requested to join
+ * @return @see e_bg_queue_apply_ack
+ */
+enum e_bg_queue_apply_ack bg_queue_join_multi(char *name, struct map_session_data *sd, std::vector <map_session_data *> list)
+{
+	if (!sd) {
+		ShowError("bg_queue_join_multi: Fatal error. Tried to join non-existent player\n.");
+		return BG_APPLY_NONE;
+	}
+
+	if (sd->sc.data[SC_ENTRY_QUEUE_APPLY_DELAY]) {
+		char buf[CHAT_SIZE_MAX];
+
+		sprintf(buf, msg_txt(sd, 339), (get_timer(sd->sc.data[SC_ENTRY_QUEUE_APPLY_DELAY]->timer)->tick - gettick()) / 1000); // You can't apply to a battleground queue for %d seconds due to recently leaving one.
+		clif_displaymessage(sd->fd, buf);
+		return BG_APPLY_NONE;
+	}
+
+	if (sd->sc.data[SC_ENTRY_QUEUE_NOTIFY_ADMISSION_TIME_OUT]) {
+		char buf[CHAT_SIZE_MAX];
+
+		sprintf(buf, msg_txt(sd, 338), ((get_timer(sd->sc.data[SC_ENTRY_QUEUE_NOTIFY_ADMISSION_TIME_OUT]->timer)->tick - gettick()) / 1000) / 60, ((get_timer(sd->sc.data[SC_ENTRY_QUEUE_NOTIFY_ADMISSION_TIME_OUT]->timer)->tick - gettick()) / 1000) % 60); // You can't apply to a battleground queue due to recently deserting a battleground. Time remaining: %d minutes and %d seconds.
+		clif_displaymessage(sd->fd, buf);
+		return BG_APPLY_NONE;
+	}
+
+	if (bg_player_is_in_bg_map(sd)) {
+		clif_displaymessage(sd->fd, msg_txt(sd, 337)); // You may not join a battleground queue when you're in a battleground map.
+		return BG_APPLY_NONE;
+	}
+
+	std::weak_ptr<s_battleground_type> weakptr = bg_search_name(name);
+	std::shared_ptr<s_battleground_type> bg;
+
+	if (weakptr.expired()) {
+		ShowWarning("bg_queue_join_multi: Could not find battleground \"%s\" requested by %s (AID: %d / CID: %d)\n", name, sd->status.name, sd->status.account_id, sd->status.char_id);
+		return BG_APPLY_INVALID_NAME;
+	} else
+		bg = weakptr.lock();
+
+	if (bg->min_lvl && sd->status.base_level < bg->min_lvl)
+		return BG_APPLY_PLAYER_LEVEL; // Level too low
+	
+	if (bg->max_lvl && sd->status.base_level > bg->max_lvl)
+		return BG_APPLY_PLAYER_LEVEL; // Level too high
+
+	if (bg_queues.empty()) {
+		std::shared_ptr<s_battleground_queue> queue = bg_queue_create(bg);
+		bool r = rnd() % 2 != 0;
+
+		if (!r) {
+			while (!list.empty() && queue->teama_members.size() < bg->required_players) {
+				struct map_session_data *sd2 = list.back();
```

Comment:
What's the reasoning behind not using `list.pop_back()` here instead of accessing through `list.back()`?

Reply:
My understanding is that `pop_back()` returns void which is why we have to use `back()` to access before hand.

Reply:
Yep, you are right. For some reason I thought the contrary.